File: Solution/剑指Offer05.替换空格/剑指Offer05.替换空格.cpp
/*
 * @Descroption: 剑指 Offer 05. 替换空格
 * @Author: EmoryHuang
 * @Date: 2021-06-28 16:42:00
 * @解题思路:
 * 简单题，替换空格，遍历或者直接 replace
 */

class Solution {
   public:
    string replaceSpace(string s) {
        while (s.find(' ') != string::npos) s.replace(s.find(' '), 1, "%20");
        return s;
    }
};

File: Solution/剑指Offer05.替换空格/剑指Offer05.替换空格.java
/*
 * @Descroption: 剑指 Offer 05. 替换空格
 * @Author: EmoryHuang
 * @Date: 2023-07-24 09:30:55
 * @Method:
 * 遍历查找空格
 */

import java.util.*;

class Solution {
    public String replaceSpace(String s) {
        StringBuilder sb = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch == ' ') {
                sb.append("%20");
            } else {
                sb.append(ch);
            }
        }
        return sb.toString();
    }
}